File: myoro_matchup/lib/module/game/domain/model/game_age_range_model.dart
import 'package:faker/faker.dart';
import 'package:flutter/foundation.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:myoro_flutter_annotations/myoro_flutter_annotations.dart';

part 'game_age_range_model.g.dart';

/// Model representing a game age range.
@immutable
@myoroModel
@JsonSerializable()
final class GameAgeRangeModel with _$GameAgeRangeModelMixin {
  /// Default constructor.
  const GameAgeRangeModel({required this.minAge, required this.maxAge});

  /// From JSON.
  factory GameAgeRangeModel.fromJson(Map<String, dynamic> json) {
    return _$GameAgeRangeModelFromJson(json);
  }

  /// Fake constructor.
  GameAgeRangeModel.fake() : minAge = faker.randomGenerator.integer(100), maxAge = faker.randomGenerator.integer(100);

  /// Minimum age.
  final int minAge;

  /// Maximum age.
  final int maxAge;

  /// To JSON.
  Map<String, dynamic> toJson() {
    return _$GameAgeRangeModelToJson(this);
  }
}

File: myoro_matchup_api_old/src/main/java/com/myoro/myoro_matchup_api/model/GameAgeRangeModel.java
package com.myoro.myoro_matchup_api.model;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;

/** Game age range model. */
@Embeddable
public class GameAgeRangeModel {
  /** Minimum age. */
  @Column(nullable = false)
  private Integer minAge;

  /** Maximum age. */
  @Column(nullable = false)
  private Integer maxAge;

  /** Default constructor. */
  public GameAgeRangeModel() {}

  /** Constructor with all fields. */
  public GameAgeRangeModel(Integer minAge, Integer maxAge) {
    this.minAge = minAge;
    this.maxAge = maxAge;
  }

  /**
   * Getter for minAge
   *
   * @return the minimum age
   */
  public Integer getMinAge() {
    return minAge;
  }

  /**
   * Getter for maxAge
   *
   * @return the maximum age
   */
  public Integer getMaxAge() {
    return maxAge;
  }

  /**
   * Setter for minAge
   *
   * @param minAge the minimum age
   */
  public void setMinAge(Integer minAge) {
    this.minAge = minAge;
  }

  /**
   * Setter for maxAge
   *
   * @param maxAge the maximum age
   */
  public void setMaxAge(Integer maxAge) {
    this.maxAge = maxAge;
  }
}
